### ISSUE TITLE
Implement Firefox Hypothesis extension

### ISSUE THREAD

**Issue by ajpeddakotla** [hypothesis/product-backlog#637]
> Users have consistently requested a Firefox extension. This card tracks this initial request.

Shipping the FF add-on supports H's goals for global usage.

From a short conversation with Rob, the primary reason this hasn't been shipped is that the manual review process was unreasonably long. After some [basic research](https://extensionworkshop.com/documentation/publish/signing-and-distribution-overview/), this concern has since been alleviated (barring a Recommended badge).

**Comment by ZiadJ** [comment]
Is this still being worked on? I'm asking because it's been years now since the issue went lingo. Firefox support is the only barrier left for mass adoption but I'm not sure I should hold my breath on it anymore.

**Comment by ZiadJ** [comment]
Is this still being worked on? I'm asking because it's been years now since the issue went limbo. Firefox support is the only barrier left for mass adoption but I'm not sure I should hold my breath on it anymore.

**Comment by robertknight** [comment]
Hi @ZiadJ - Still on the roadmap but we haven't been actively working on it as we've had other priorities for various reasons. You can build a Firefox extension from code in the [browser-extension](https://github.com/hypothesis/browser-extension) repository but there is probably work needed to adapt to the latest Firefox releases (WebExtensions has been undergoing quite a lot of changes) and on our side to actually get it into production.

**Comment by klemay** [comment]
@ZiadJ we do have a bookmarklet that works with Firefox which you could use in the meantime - see Step 2 here: https://web.hypothes.is/start/

**Comment by grahamperrin** [comment]
> probably work needed to adapt to the latest Firefox releases

FWIW with Waterfox Classic 56.2.12 I can run [the extension for Chromium-based browsers](https://chrome.google.com/webstore/detail/hypothesis-web-pdf-annota/bjfhmglciegochdpefhhlphglcehbmek) through [Chrome Store Foxified](https://addons.mozilla.org/addon/chrome-store-foxified/) and get this: 

![image](https://user-images.githubusercontent.com/192271/61596263-eee90b00-abf8-11e9-9e82-467d5ffcc771.png)

– and authentication apparently succeeds but then the sidebar continues to show the prompt to log in. 

The not logged in (NLI) view does, at least, show annotations, for example: 

![image](https://user-images.githubusercontent.com/192271/61596317-923a2000-abf9-11e9-8037-1b457a4e3b2a.png)

### Suggestion

The badge should be disabled by default following installation of the extension. 

Rationale: privacy.

**Comment by grahamperrin** [comment]
> … a bookmarklet that works with Firefox …

Whilst it's nice to have the bookmarklet, there's the unavoidable frustration of bookmarklets of this type (not solely Hypothesis) not working with various sites. 

[Note that the bookmarklet doesn't work on sites with restrictive CSP · Issue #209 · hypothesis/vision](https://github.com/hypothesis/vision/issues/209) (2016, archived)

So for example https://twitter.com/judell/status/1152257422125223936 again, in a different Waterfox Classic profile, after clicking the bookmarklet: 

![image](https://user-images.githubusercontent.com/192271/61596379-ad595f80-abfa-11e9-8d98-2e72d7f4b56f.png)

– to the right, no sidebar.

**Comment by judell** [comment]
> FWIW with Waterfox Classic 56.2.12 I can run the extension for Chromium-based browsers through Chrome Store Foxified

@grahamperrin that's very interesting. I also use Waterfox to run an orphan ftp add-on, did not know about Chrome Store Foxified, which is evidently discontinued but also runs as an orphan in Waterfox.

You got farther than I did! If I could reproduce what you've done to get the Hypothesis sidebar to open in Waterfox by way of Chrome Store Foxified, I might be able to suss out your auth issue. But I can't. Are you able to reproduce the situation in which the sidebar did load, and you were able to (albeit not stickily) authenticate?

**Comment by grahamperrin** [comment]
> Are you able to reproduce the situation in which the sidebar did load, and you were able to (albeit not stickily) authenticate?

Yes, with home-built Waterfox Classic 56.2.12 and a new profile on FreeBSD-CURRENT. 

Join me in IRC if you like (directions: https://web.hypothes.is/help/how-can-i-get-access-to-the-code-or-talk-to-hypothesis-developers/#hkb), I'm connected via Riot/Matrix. I'll be snoozing whilst connected so just fire away and I'll pick up reasonably soon. 

Thanks

**Comment by lidel** [comment]
FYSA Firefox 69 will fix bookmarklets on sites with restrictive CSP: 
https://bugzilla.mozilla.org/show_bug.cgi?id=1478037

**Comment by rubaboo** [comment]
Leaving a comment just to watch this conversation. Hope it stays open until the extension is released.

**Comment by TTimo** [comment]
An extension to get past the CSP problems would be much welcome.

**Comment by ankostis** [comment]
FYI, on Android you may switch to [Firefox-nightly("fenix)](https://play.google.com/store/apps/details?id=org.mozilla.fenix) which permits the installation of ANY add-on, when following [the official procedure](http://blog.mozilla.org/addons/2020/09/29/expanded-extension-support-in-firefox-for-android-nightly/), including the [the "unofficial bookmarklet" from Hernawan](https://addons.mozilla.org/en-US/firefox/addon/hypothes-is-bookmarklet/) which i tested it and work flawlessly even in restrictive-CSP sites like GitHub.

I tested also the ["unofficial-hypothesis" add-on](https://addons.mozilla.org/en-US/firefox/addon/unofficial-hypothesis/) but it didn't work due to diegodlh/unofficial-hypothesis-extension#4.

**Comment by citydelver** [comment]
Hello. This request exists since 2018 (or perhaps longer). Did any development happen?

**Comment by robertknight** [comment]
We still have a Firefox extension "in development" (as in, you can check out this repository and build one) but we've never gotten to the point of actually shipping it as something you can download from our website or addons.mozilla.org for various reasons. I'm afraid I can't make any promises about when that will happen, except to say that it is something we would still like to do.

**Comment by citydelver** [comment]
Got it. We still miss it, though. So instead of asking "when", may we ask you "who" (who could do/is doing this), so that we can positively encourage him/her? :)

**Comment by Facni** [comment]
It still WIP?
